### ISSUE TITLE
Reloading table data with non-ajax datasource

### ISSUE THREAD

**Issue by michahell** [DataTables/DataTables#585]
Since this is somehow extremely hard to do, I'm adding in another API method to do this,
based on the API.ajax.reload() functionality. I'm working on a Fork so that a minified version becomes available, but here's my quick fix:

```javascript
  /**
   * Reload tables from their javascript data source. Fuck the Ajax only reload!
   * Note that this function will
   * automatically re-draw the table when the remote data has been loaded.
   *
   * @param {boolean} [reset=true] Reset (default) or hold the current paging
   *   position. A full re-sort and re-filter is performed when this method is
   *   called, which is why the pagination reset is the default action.
   * @returns {DataTables.Api} this
   */
  _api_register( 'js.reload()', function ( data, resetPaging ) {
    return this.iterator( 'table', function (settings) {
      
      _fnProcessingDisplay( settings, true );
      _fnClearTable( settings );
      
      for ( var i=0, ien=data.length ; i<ien ; i++ ) {
        _fnAddData( settings, data[i] );
      }

      _fnReDraw( settings, resetPaging===false );
      _fnProcessingDisplay( settings, false );
    } );
  } );
```

**Comment by DataTables** [comment]
Could you clarify why you found this hard to do? 'table.clear().rows.add( data ).draw()' would clear the table, add data and redraw.

**Comment by michahell** [comment]
Hmm, that works perfectly. 
I meant to say, it's extremely hard to **find** how to do this easily. I've searched high and low for a simple 'reload', 'refresh', 'new data' method but could not find one. There is no documentation on how to refresh/reload any other data source then Ajax, which explains the many stack overflow and other forum posts I did found using google. It would greatly help if this single line to reload an entire table's data could be found in the documentation!

Issue resolved, fork deleted. Thank you for responding.

**Comment by DataTables** [comment]
Hi,

To my mind, the problem with calling it a 'refresh' or 'reload' of the Javascript sourced data is that it is not reloading from the same source (which is what the Ajax call does) - but rather you are feeding the data into the API in order to add it to the table. There can't really be a _reload_ of the data since DataTables doesn't know where the data has come from and therefore can't request new data.

Instead, you clear the table and then add new rows. That is exactly what you did using the private methods. What I don't quite get is why you didn't just use the public API to achieve exactly that :-)

**Comment by michahell** [comment]
Aha. I see, and I agree. :)
I guess people (like me) start looking for a reload method because somehow it is the best way to (conceptually) describe what you want to achieve finally. 
I also think that 'reload' like methods are so ingrained (at least for me) that because I did not find them, I did not think about just clearing, inserting and updating the table as a solution. 
For me, I think it is also because I thought that maybe some 'object pooling' system was used.
If it did, then clearing, inserting and then redrawing would circumvent the pooling system and be less efficient / fast.
